content/articles/files-and-exeptions-in-python/index.md

```diff
@@ -0,0 +1,265 @@
+
+Python uses the *try-except* block to control exceptions. A try-except block informs Python how to act when an exception emerges. Our programs will continue to execute even if things go wrong.
+
+### Handling the ZeroDivisionError exception
+
+Let’s run a program that divides a number by zero. We absolutely know it is impractical, but let’s see what Python will do.
+
+```python
+
+print(6/0)
+
+```
+
+When we run the above code Python gives the following trace-back:
+
+```python
+
+Traceback (most recent call last):
+
+  File “C:\Users\ADMIN\PycharmProject\pythonProject\main.py”, line 1, in <module>
+
+ print(6/0)
+
+ZeroDivisionError: division by zero
+
+```
+
+Since Python cannot divide a number by zero, it reports an error in the trace-back, *ZeroDivisionError*, which is an exception object. This kind of object responds to a scenario where Python can’t do what we asked it to.
+
+If you think of an error occurring use the *try-except* block to control the exception that may be raised.
+
+To handle the ZeroDivisionError exception use a try-except block like this:
+
+```python
+
+try:
+
+ print(6/0)
+
+except ZeroDivisionError:
+
+ print(“You can’t divide by zero!”) # You can’t divide by zero!
+
+```
+
+### Handling the FileNotFoundError exception
+
+Errors will always arise when working with files that are missing. Python may fail to get file if you have written the wrong spelling of filename, or the file never exists. We handle this situation by applying the try-except block. For example, the program below tries to read a file that doesn’t exist on my computer.
+
+```python
+
+filename = ‘John.txt’
+
+with open(filename) as f_obj:
+
+ contents = f_obj.read()
+
+```
+
+ Since Python can not read a file that does not exist, it raises an exception:
+
+```python
+
+Traceback (most recent call last):
+
+  File “C:\Users\ADMIN\PycharmProject\pythonProject\main.py”, line 2, in <module>
+
+ with open(filename) as f_obj:
+
+FileNotFoundError: [Errno 2] No such file or directory: ‘john.txt’
+
+```
+
+Since Python can not find the file, we are opening it creates an exception that is the *FileNotFoundError* exception. In this example, the **open()** function creates the error. To solve this error, use the try block just before the line, which involves the **open()** function:
+
+```python
+
+filename = ‘John.txt’
+
+try:
+
+ with open(filename) as f_obj:
+
+       contents = f_obj.read()
+
+except FileNotFoundError:
+
+ msg = “Sorry, the file “+ filename + “does not exist.”
+
+ print(msg) # Sorry, the file John.txt does not exist.
+
+```
+### Conclusion
+
+In this article, we have learnt how to:
+
+ - Read from a file
+
+ - Work with a File’s Contents
+
+ - Write to a File
+
+ - Handle the ZeroDivisionError Exception
+
+ - Handle the FileNotFoundError Exception
+
```

Comment:
Add a `Further reading`/`References` section where you add external links that helps reader gain more knowledge on this topic